src/components/Map/components/CardSearchDetails.tsx

```diff
@@ -19,18 +25,31 @@ const CardSearchDetails = ({ result, onClick, onClickClose }: SearchResult) => {
   const { distance } = result.addressDetails?.networkDetails?.network || {};
   const { isEligible } = result.addressDetails?.networkDetails || {};
 
-  const onClickHandler = (ev: React.MouseEvent<HTMLElement>) => {
-    const returnVal =
-      (typeof onClick === 'function' && onClick(result)) || undefined;
-    ev.stopPropagation();
-    return returnVal;
-  };
+  const [contactFormVisible, setContactFormVisible] = useState(false);
 
-  const onCloseHandler = () => {
-    if (onClickClose) {
-      onClickClose(result);
-    }
-  };
+  const onClickHandler = useCallback(
+    (evt: React.MouseEvent<HTMLElement>) => {
+      evt.stopPropagation();
+      const returnVal =
+        (typeof onClick === 'function' && onClick(result)) || undefined;
+      return returnVal;
+    },
+    [onClick, result]
+  );
+
+  const onCloseHandler = useCallback(
+    (evt: React.MouseEvent<HTMLElement>) => {
+      evt.stopPropagation();
```

Comment:
what's the difference between stopPropagation and preventDefault ?

Reply:
The preventDefault is for don't execute the default event. The stop propagation is to prevent the capturing and bubbling phases.
Ex. With a onClick event on link over a map: 
- With preventDefault, the onClick event is apply without the default event (follow the target href link), but after the link's onClick event, the map's onClick event is also executed (and all the event of each parents). 
- With stopPropagation, only the link 's onClick event is executed.